src/cobra/util/solver.py

```diff
@@ -313,6 +315,49 @@ def choose_solver(
     return solver
 
 
+def check_solver(obj):
+    """Check whether the chosen solver is valid.
+
+    Check whether chosen solver is valid and also warn when using
+    a specialized solver. Will return the optlang interface for the
+    requested solver.
+
+    Parameters
+    ----------
+    obj : str or optlang.interface or optlang.interface.Model
+        The chosen solver.
+
+    Raises
+    ------
+    SolverNotFound
+        If the solver is not valid.
+    """
+    not_valid_interface = SolverNotFound(
+        "%s is not a valid solver interface. Pick from %s." % (obj, list(solvers))
```

Comment:
Could use f-strings now.

```suggestion
        f"{obj} is not a valid solver interface. Pick one from {', '.join(solvers)}."
```